alchemist/alchemist-cognitive-agents/src/main/kotlin/it/unibo/alchemist/model/implementations/actions/Utils.kt

```diff
@@ -37,3 +40,27 @@ fun <T, P> Sequence<P>.discardUnsuitablePositions(
          */
         environment !is PhysicsEnvironment<T, P, *, *> || environment.canNodeFitPosition(pedestrian, it)
     }
+
+/**
+ * A delegate allowing to lazily initialise a mutable variable (= var).
+ */
+class LazyMutable<T>(private val initializer: () -> T) : ReadWriteProperty<Any?, T> {
```

Comment:
Reify this in its own file. I do not like having "Utils" files stuffed with a lot of stuff. And by the way this is so generic that I wonder why it's not in library... is there some side effect to its existence and use?

Reply:
Moved to its own file. There are no side effects as far as I know, probably it's not in the library because you can have a `lateinit var`, even though `lateinit` doesn't allow you to specify a lazy initialiser, which is why I created that class.